lib/remote/configstageshandler.cpp

```diff
@@ -128,19 +136,18 @@ void ConfigStagesHandler::HandlePost(
 		if (reload && !activate)
 			BOOST_THROW_EXCEPTION(std::invalid_argument("Parameter 'reload' must be false when 'activate' is false."));
 
-		std::unique_lock<std::mutex> lock(ConfigPackageUtility::GetStaticPackageMutex());
```

Comment:
This mutex is used in other places too, have you checked that it's safe to remove here?

Reply:
> This mutex is used in other places too

It is also allowed concurrent access everywhere it is still used.

> have you checked that it's safe to remove here?

There is no concurrent access here anymore, so this lock is not needed.

Reply:
There are other places than this one that also lock on `ConfigPackageUtility::GetStaticPackageMutex()`:
* `ConfigObjectUtility::CreateStorage()`
* `ConfigPackagesHandler::HandleGet(...)`
* `ConfigPackagesHandler::HandlePost(...)`
* `ConfigPackageUtility::TryActivateStageCallback(...)`

This PR (with fixed synchronization of `m_RunningPackageUpdates` will only prevent this function from running concurrently with the last one in that list. The lock could still be required for one of the other three functions running concurrently.

(I'm not implying that there is a problem with removing the lock here, but one has to carefully check if it's safe to do so.)